### PR TITLE
Docs(Room): Update JavaDocs for Room

### PR DESCRIPTION
**Pull request by DrWala**
Increases JavaDoc coverage for the Room sections of the code.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java [hunk 1]
```diff
@@ -60,8 +60,10 @@
     private final EditRoomDescriptor editRoomDescriptor;
 
     /**
-     * @param index              of the room in the filtered room list to edit
-     * @param editRoomDescriptor details to edit the room with
+     * Creates an EditRoomCommand to edit the room at the specified {@code Index}.
+     *
+     * @param index              of the room in the filtered room list to edit.
```
Comment (stein414) [inline src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java:65]: Try not to use the parameter in the sentence.
Capitalise the sentence

File: src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java [hunk 2]
```diff
@@ -35,12 +35,22 @@
 
     /**
      * Creates an AddRoomCommand to add the specified {@code Room}
+     *
+     * @throws NullPointerException if {@code Room} is null
```
Comment (stein414) [inline src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java:39]: Missing fullstop

File: src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java [hunk 3]
```diff
@@ -35,12 +35,22 @@
...
      */
     public AddRoomCommand(Room room) {
         requireNonNull(room);
         this.toAdd = room;
     }
 
+    /**
+     * Executes the AddRoomCommand with the specified {@code Model}
```
Comment (stein414) [inline src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java:47]: Missing fullstop.

File: src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java [hunk 4]
```diff
@@ -35,10 +35,26 @@
...
 
+    /**
+     * Executes the DeleteRoomCommand on the {@code Room} at the specified {@code Index} on the provided {@code Model}.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return A {@code CommandResult} with the {@code Room} that was deleted.
+     * @throws CommandException If the room being deleted at the specified {@code Index} does not exist.
+     * @throws CommandException If the room being deleted at the specified {@code Index} has been allocated to a
+     *                          {@code Resident}.
+     * @throws CommandException If the room being deleted at the specified {@code Index} has any {@code Issue}
```
Comment (stein414) [inline src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java:55]: Missing fullstop.

File: src/main/java/seedu/address/logic/parser/room/AddRoomCommandParser.java [hunk 5]
```diff
@@ -26,15 +26,17 @@
 public class AddRoomCommandParser implements Parser<AddRoomCommand> {
 
     /**
-     * Parses the given {@code String} of arguments in the context of the AddRoomCommand
-     * and returns an AddRoomCommand object for execution.
+     * Parses the given {@code String} of arguments in the context of the AddRoomCommand and returns an AddRoomCommand
+     * object for execution.
      *
+     * @param userInput The command {@code String} entered by the user
```
Comment (stein414) [inline src/main/java/seedu/address/logic/parser/room/AddRoomCommandParser.java:32]: Missing fullstops

File: src/main/java/seedu/address/logic/parser/room/DeleteRoomCommandParser.java [hunk 6]
```diff
@@ -16,6 +16,8 @@
      * Parses the given {@code String} of arguments in the context of the DeleteRoomCommand
      * and returns a DeleteRoomCommand object for execution.
      *
+     * @param userInput The command {@code String} entered by the user
```
Comment (stein414) [inline src/main/java/seedu/address/logic/parser/room/DeleteRoomCommandParser.java:19]: Missing fullstop

File: src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java [hunk 7]
```diff
@@ -29,6 +29,8 @@
      * Parses the given {@code String} of arguments in the context of the EditRoomCommand
      * and returns an EditRoomCommand object for execution.
      *
+     * @param userInput The command {@code String} entered by the user
```
Comment (stein414) [inline src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java:32]: Missing fullstop

File: src/main/java/seedu/address/logic/parser/room/FindRoomCommandParser.java [hunk 8]
```diff
@@ -15,10 +15,12 @@
      * Parses the given {@code String} of arguments in the context of the FindRoomCommand
      * and returns a FindRoomCommand object for execution.
      *
+     * @param userInput The command {@code String} entered by the user
```
Comment (stein414) [inline src/main/java/seedu/address/logic/parser/room/FindRoomCommandParser.java:18]: Missing fullstop

File: src/main/java/seedu/address/model/room/UniqueRoomList.java [hunk 9]
```diff
@@ -29,15 +29,24 @@
             FXCollections.unmodifiableObservableList(internalList);
 
     /**
-     * Returns true if the list contains an equivalent room as the given argument.
+     * Checks the list contains an equivalent {@code Room} as the given argument.
+     *
+     * @param toCheck {@code Room} to check if it is contained within the {@link UniqueRoomList#internalList}
```
Comment (stein414) [inline src/main/java/seedu/address/model/room/UniqueRoomList.java:34]: Missing fullstop

File: src/main/java/seedu/address/model/room/UniqueRoomList.java [hunk 10]
```diff
@@ -29,15 +29,24 @@
...
         requireNonNull(toCheck);
         return internalList.stream().anyMatch(toCheck::isSameRoom);
     }
 
     /**
-     * Returns true if the list contains an equivalent room as the given argument.
+     * Checks the list contains a {@code Room} which has the {@code RoomNumber} of the given argument.
+     *
+     * @param toCheck {@code RoomNumber} to check if there is a room contained within the
+     *                {@link UniqueRoomList#internalList} that matches it
```
Comment (stein414) [inline src/main/java/seedu/address/model/room/UniqueRoomList.java:47]: Missing fullstop

**Review by stein414 — CHANGES_REQUESTED** [review]
A few small formatting issues.

File: src/main/java/seedu/address/model/room/UniqueRoomList.java (hunk 10 above)
Comment (DrWala) [reply src/main/java/seedu/address/model/room/UniqueRoomList.java]: Fixed

File: src/main/java/seedu/address/model/room/UniqueRoomList.java (hunk 9 above)
Comment (DrWala) [reply src/main/java/seedu/address/model/room/UniqueRoomList.java:34]: Fixed

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java (hunk 1 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java]: Fixed

File: src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java (hunk 2 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java]: Fixed

File: src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java (hunk 3 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java]: Fixed

File: src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java (hunk 4 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java:55]: Fixed

File: src/main/java/seedu/address/logic/parser/room/AddRoomCommandParser.java (hunk 5 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/parser/room/AddRoomCommandParser.java]: Fixed

File: src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java (hunk 7 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java]: Fixed

File: src/main/java/seedu/address/logic/parser/room/FindRoomCommandParser.java (hunk 8 above)
Comment (DrWala) [reply src/main/java/seedu/address/logic/parser/room/FindRoomCommandParser.java]: Fixed

File: src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java [hunk 11]
```diff
@@ -35,12 +35,22 @@
...
         requireNonNull(room);
         this.toAdd = room;
     }
 
+    /**
+     * Executes the AddRoomCommand with the specified {@code Model}.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return A {@code CommandResult} with the {@code Room} that was added.
+     * @throws CommandException If the room being added is a duplicate of one already in the model.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/AddRoomCommand.java:51]: Should this throw NullPointerException?

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java [hunk 12]
```diff
@@ -60,8 +60,10 @@
     private final EditRoomDescriptor editRoomDescriptor;
 
     /**
-     * @param index              of the room in the filtered room list to edit
-     * @param editRoomDescriptor details to edit the room with
+     * Creates an EditRoomCommand to edit the room at the specified {@code Index}.
+     *
+     * @param index              Of the room in the filtered room list to edit.
+     * @param editRoomDescriptor Details to edit the room with.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java:66]: Should this throw NullPointerException?

File: src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java [hunk 13]
```diff
@@ -35,10 +35,26 @@
...
+     *                          associated with it.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java:56]: Should this throw NullPointerException?

File: src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java (hunk 4 above)
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/DeleteRoomCommand.java:55]: Do you want to combine these into one @throws parameter? Saw this in other places.

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java [hunk 14]
```diff
@@ -122,8 +134,11 @@ public CommandResult execute(Model model) throws CommandException {
     }
 
     /**
-     * Creates and returns a {@code Room} with the details of {@code roomToEdit}
-     * edited with {@code editRoomDescriptor}.
+     * Creates and returns a {@code Room} with the details of {@code roomToEdit} edited with {@code editRoomDescriptor}.
+     *
+     * @param roomToEdit         The {@code Room} that is being edited
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java:139]: Missing fullstop?

File: src/main/java/seedu/address/logic/commands/room/FindRoomCommand.java [hunk 15]
```diff
@@ -26,10 +26,24 @@
...
         this.predicate = predicate;
     }
 
+    /**
+     * Executes the FindRoomCommand on the provided {@code Model} which will ensure that only rooms matching the
+     * provided {@code RoomNumberOrTagsContainsKeywordsPredicate} will be returned in
+     * {@link Model#getFilteredRoomList()}.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return A {@code CommandResult} with the number of {@code Room} objects that will be listed.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/FindRoomCommand.java:45]: Should this throw NullPointerException?

File: src/main/java/seedu/address/logic/commands/room/ListRoomCommand.java [hunk 16]
```diff
@@ -21,6 +21,11 @@
     private static final Logger logger = LogsCenter.getLogger(ListRoomCommand.class);
 
 
+    /**
+     * Executes the ListRoomCommand on the provided {@code Model} list all the rooms.
+     * @param model {@code Model} which the command should operate on.
+     * @return A {@code CommandResult} with the number of {@code Room} objects that will be listed.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/commands/room/ListRoomCommand.java:27]: Should this throw NullPointerException?

File: src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java [hunk 17]
```diff
@@ -29,7 +29,9 @@
      * Parses the given {@code String} of arguments in the context of the EditRoomCommand
      * and returns an EditRoomCommand object for execution.
      *
-     * @throws ParseException if the user input does not conform the expected format
+     * @param userInput The command {@code String} entered by the user.
+     * @return The parsed {@code EditRoomCommand}.
+     * @throws ParseException if the user input does not conform the expected format.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/logic/parser/room/EditRoomCommandParser.java:34]: Should this throw NullPointerException?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 18]
```diff
@@ -212,20 +231,38 @@ public void addRoom(Room room) {
...
 
         statefulAddressBook.removeRoom(target);
     }
 
+    /**
+     * Replaces a given {@code Room} with another.
+     *
+     * @param target     {@code Room} to replace.
+     * @param editedRoom {@code Room} to replace {@code target} with.
+     * @throws NullPointerException If {@code RoomNumber} is null.
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/ModelManager.java:251]: Will it throw if room is null?

File: src/main/java/seedu/address/model/room/Room.java [hunk 19]
```diff
@@ -42,7 +48,14 @@ public Room(RoomNumber roomNumber, RoomType roomType, IsOccupied isOccupied, Set
     }
 
     /**
-     * Every field must be present and not null.
+     * Constructs a Room with the default {@code IsOccupied} value of "No".
+     * This constructor should be used in most cases, i.e creating a room using the {@code AddRoomCommand} or
+     * {@code EditRoomCommand}
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/room/Room.java:53]: Missing fullstop?

File: src/main/java/seedu/address/model/room/UniqueRoomList.java [hunk 20]
```diff
@@ -90,6 +108,12 @@ public void remove(Room toRemove) {
         FXCollections.sort(internalList);
     }
 
+    /**
+     * Replaces the entire {@code UniqueRoomList} with the provided {@code replacement}
```
Comment (benedictkhoomw) [inline src/main/java/seedu/address/model/room/UniqueRoomList.java:112]: Missing fullstop?

**Review by benedictkhoomw — CHANGES_REQUESTED** [review]
Just a few nits to fix.

**Review by stein414 — APPROVED** [review]
LGTM

**Review by benedictkhoomw — APPROVED** [review]
LGTM.

**Pull request merged**